### DISCUSSION TITLE
Add  product to order with customer relation.

### DISCUSSION THREAD

**Discussion by ivsaneesh** [vendurehq/vendure#1583] [Q&A]
I was trying to add product to order, but I was not able to create an order with relation with customer.
How can I  add an item to the cart and how the order is related to user.

**Comment by michaelbromley** [comment]
Hi,

Some questions:

- What exact mutation did you execute?
- What was the result returned?

**Comment by ivsaneesh** [comment]
i used addItemToOrder 

mutation {
  addItemToOrder(quantity: 1, productVariantId: 7) {
    __typename
  }
}

and the result was

{
  "data": {
    "addItemToOrder": {
      "__typename": "Order"
    }
  }
}

![Screenshot 2022-05-26 093307](https://user-images.githubusercontent.com/34297370/170413556-49ed5577-ba46-45f6-8e25-804e8fe70d38.jpg)

**Reply by ivsaneesh** [reply]
How the order is related to customer.
i don't know how to connect the order with customer.

**Reply by michaelbromley** [reply]
Ok thanks.

So first of all, you can also select the fields of the resulting Order like this:
```graphql
mutation {
  addItemToOrder(quantity: 1, productVariantId: 7) {
    ...on Order {
      id
      state
      totalWithTax
      totalQuantity
      lines {
        id
        linePriceWithTax
      }
    }
    ...on ErrorResult {
      errorCode
      message
    }
  }
}
```

> i don't know how to connect the order with customer.

If the customer is not currently signed in or does not have an account, then you'd use the [setCustomerForOrder](https://www.vendure.io/docs/graphql-api/shop/mutations/#setcustomerfororder) mutation which will create a new Customer entity to go with the Order.
